subworkflows/local/profiling.nf

```diff
@@ -129,7 +131,37 @@ workflow PROFILING {
         ch_multiqc_files       = ch_multiqc_files.mix( KRAKEN2_KRAKEN2.out.report )
         ch_versions            = ch_versions.mix( KRAKEN2_KRAKEN2.out.versions.first() )
         ch_raw_classifications = ch_raw_classifications.mix( KRAKEN2_KRAKEN2.out.classified_reads_assignment )
-        ch_raw_profiles        = ch_raw_profiles.mix( KRAKEN2_KRAKEN2.out.report )
+        ch_raw_profiles        = ch_raw_profiles.mix(
+            KRAKEN2_KRAKEN2.out.report
+                // Set the tool to be strictly 'kraken2' instead of potentially 'bracken' for downstream use.
+                .map { meta, report -> [meta + [tool: 'kraken2'], report]}
+        )
+
+    }
+
+    if ( params.run_kraken2 && params.run_bracken ) {
+
+        def ch_input_for_bracken = KRAKEN2_KRAKEN2.out.report
+            .filter { meta, report -> meta['tool'] == 'bracken' }
+
+        if (params.kraken2_save_minimizers) {
+            ch_input_for_bracken = KRAKEN2_STANDARD_REPORT(ch_input_for_bracken).report
+        }
+
+        ch_input_for_bracken = ch_input_for_bracken
+            .combine(
+                databases.filter { meta, db ->
```

Comment:
Why re-access `databases` as input channel, and not a 

```
ch_input_for_profiling.bracken
```

?? given you combine again here, and use the multiMap again?

(it's Friday though so might be my brain not working properly)